src/stack-cli.ts

```diff
@@ -0,0 +1,110 @@
+import * as fs from "fs";
+import { devNull } from "os";
+import type { ExecOptions } from "@actions/exec";
+import * as realExec from "@actions/exec";
+
+import type { StackPath } from "./parse-stack-path";
+import { parseStackPath } from "./parse-stack-path";
+import type { StackQuery } from "./parse-stack-query";
+import { parseStackQuery } from "./parse-stack-query";
+
+export interface ExecDelegate {
+  exec: (
+    command: string,
+    args: string[],
+    options?: ExecOptions,
+  ) => Promise<number>;
+}
+
+export class StackCLI {
+  private debug: boolean;
+  private globalArgs: string[];
+  private execDelegate: ExecDelegate;
+
+  constructor(
+    stackYaml: string,
+    args: string[],
+    debug?: boolean,
+    execDelegate?: ExecDelegate,
+  ) {
+    this.debug = debug ?? false;
+    this.execDelegate = execDelegate ?? realExec;
```

Comment:
If `execDelegate` is used only for testing, I think it's better to use Jest to mock `@actions/exec` in the test module and remove the little indirection here around the `exec` import.

Reply:
Is that our preferred style? I saw this "delegate" stuff going on in the GitHub libraries I was looking at and it seemed nice. Both as a way to mock but also to constrain the interface further (the `interface`s we use here are fewer functions and arguments than the real things).

I'll give it a try.